### PR TITLE
Stellar: Export Secret Key screen

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

First up: @patrickxb or @mlsteele, are you okay with moving `exportSecretKeyLocal` to `enabled-calls.json` now?  It's used by the frontend for this PR.

Redux security considerations: Frontenders, I decided to use a `HiddenString` for the Stellar secret key value in the redux store, and to clear that `HiddenString` out from the store as soon as the export dialog is closed.  Sound right?

### CODE REVIEW

**Comment by mlsteele** [comment]
@cjb yep sounds good

**Comment by cjb** [comment]
@zanderz Thanks -- is that a thumb? :) Pinentry's switched over to the non-passphrase version now.

File: shared/actions/wallets.js [hunk 1]
```diff
@@ -121,6 +133,7 @@ function* walletsSaga(): Saga.SagaGenerator<any, any> {
     [WalletsGen.selectAccount, WalletsGen.linkedExistingAccount],
     navigateToAccount
   )
+  yield Saga.safeTakeEveryPure(WalletsGen.exportSecretKey, exportSecretKey, exportSecretKeySuccess)
```
Comment (chrisnojima) [inline shared/actions/wallets.js]: likely better to use safeTakeEveryPurePromise

File: shared/reducers/wallets.js [hunk 2]
```diff
@@ -18,6 +18,10 @@ export default function(state: Types.State = initialState, action: WalletsGen.Ac
       return state.setIn(['assetsMap', action.payload.accountID], I.List(action.payload.assets))
     case WalletsGen.paymentsReceived:
       return state.setIn(['paymentsMap', action.payload.accountID], I.List(action.payload.payments))
+    case WalletsGen.secretKeyReceived:
+      return state.setIn(['secretKeyMap', action.payload.accountID], action.payload.secretKey)
+    case WalletsGen.secretKeySeen:
+      return state.set('secretKeyMap', I.Map())
```
Comment (chrisnojima) [inline shared/reducers/wallets.js]: why have this be a map and set it based on account but clear the entire map? likely this can just be one value right

Comment (chrisnojima) [reply shared/reducers/wallets.js]: if you switch accounts it can just clear it

File: shared/constants/wallets.js [hunk 3]
```diff
@@ -163,6 +164,9 @@ const getFederatedAddress = (state: TypedState, accountID?: Types.AccountID) =>
   return username && account.isDefault ? `${username}*keybase.io` : ''
 }
 
+const getSecretKey = (state: TypedState, accountID: Types.AccountID) =>
```
Comment (chrisnojima) [inline shared/constants/wallets.js]: likely better to return a string always so you don't have to check and call stringValue() later

File: shared/wallets/export-secret-key/container.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+export type OwnProps = {
+  accountID: Types.AccountID,
+}
+
+const mapStateToProps = (state: TypedState, {routeProps}) => {
+  const accountID = routeProps.get('accountID')
+  const secretKey = Constants.getSecretKey(state, accountID)
+  return {
+    accountID,
+    secretKey: secretKey && secretKey.stringValue(),
```
Comment (chrisnojima) [inline shared/wallets/export-secret-key/container.js]: then here its just `secretKey,`

File: shared/wallets/export-secret-key/index.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    marginBottom: globalMargins.medium,
+  },
+  secretKeyContainer: {
+    marginBottom: globalMargins.medium,
+    maxWidth: 272,
+    width: '100%',
+  },
+})
+
+const containerStyle = platformStyles({
```
Comment (chrisnojima) [inline shared/wallets/export-secret-key/index.js]: this should go into styles

**Comment by cjb** [comment]
@chrisnojima Thanks! Made all of these changes.

**Pull request merged**